poems_parser.py

```diff
@@ -1,67 +1,92 @@
-    # create folder
-    path = os.path.dirname(os.path.abspath(args.output_file))
-    if not os.path.exists(path):   # in python3.7+ you can avoid this check, just:
-        os.makedirs(path)          # os.makedirs(path, exist_ok=True)
 
-    # There is a way to make it much more simple
-    # with open(args.output_file, "w") as f:
-    #     for p in poem:
-    #         f.write(p.string.strip() + '\n')
+# парсить посилання на вірш
+def parse_poem_by_url(url):
+    poem_page = get_resource(url)
+    soup = BeautifulSoup(poem_page, 'html.parser')
+    poem_header = soup.find('span', class_="text-head2").text
+    poem_text = soup.find('p', align='left').text
+    poem = (poem_header + '\n' + poem_text + '\n')
+    return poem
 
-    # open file
-    file_to_write = codecs.open(args.output_file, "w", "utf-8")
-    # write to file
-    for p in poem:
-        file_to_write.write(p.string.strip() + '\n')
-    # close the file
-    file_to_write.close()
 
-    # you don't need read this data from file, just:
-    # print('Amount of poems =', len(poem))
+def print_whole_poem(poem_url):
+    poem = parse_poem_by_url(poem_url)
+    print(poem)
 
-    with codecs.open(args.output_file, "r", "utf-8") as file:
-        print('Amount of poems =', len(file.readlines()))
 
-    print("Absolute path = ", path)
+def main():
+    # parse the poems titles
+    url = 'http://taras-shevchenko.in.ua/'
+    poem = parse_page(get_resource(url + 'virshi-shevchenka.html'), url)
+    # parse the arguments
+    parser = argparse.ArgumentParser(description='My arsg parser')
+    parser.add_argument('--output_file', action="store", type=str)
+    parser.add_argument('--sort_poems', '-name', '-years', action='store', nargs=1)
+    parser.add_argument('--sort_order', '-asc', '-desc', action='store', nargs=1)
+    parser.add_argument('--display_novel', action='store')
+    args = parser.parse_args()
+    reverse_sort_order = False
+    if args.sort_order and args.sort_order[0] == 'desc':
+        reverse_sort_order = True
+    if args.output_file:
+        # create folder
+        path = os.path.dirname(os.path.abspath(args.output_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        # open file
+        with open(args.output_file, "w", encoding='utf-8') as f:
+            for p in poem:
+                f.write((p[0].strip() + ' ' + p[1].strip()) + '\n')
+        print('Count of poem =', len(poem))
+        print("Absolute path = ", path)
+    elif args.sort_poems and args.sort_poems[0] == 'name':
+        sorted_poems = sorted(poem, key=operator.itemgetter(0), reverse=reverse_sort_order)
+        for p in sorted_poems:
+            print(p[0].strip() + ' ' + p[1])
+    elif args.sort_poems and args.sort_poems[0] == 'years':
+        sorted_poems = sorted(poem, key=operator.itemgetter(1), reverse=reverse_sort_order)
+        for p in sorted_poems:
+            print(p[0].strip() + ' ' + p[1])
```

Comment:
Please work according to the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle:
```suggestion
    elif args.sort_poems:
        sort_order = True if args.sort_order == 'desc' else False
        if args.sort_poems == 'name':
            column_index = 0
        elif args.sort_poems == 'years':
            column_index = 1
        sorted_poems = sorted(poem, key=operator.itemgetter(column_index), reverse=sort_order)
        for p in sorted_poems:
            print(p[0].strip() + ' ' + p[1])
```